### PR TITLE
Default broker selection in CR

### LINKED ISSUE #175: Broker selection fields on install
**Problem**

0.14 will come with two brokers:
* channel based 
* multitenant channel broker

There are now different artifacts, that we need to take into account.

We have two convenient release yamls:
* [`eventing.yaml`](https://github.com/knative/eventing/blob/master/hack/release.sh#L35): this have the the `channel` version
* [`eventing-mt.yaml`](https://github.com/knative/eventing/blob/master/hack/release.sh#L36): this will have the multitenant version

However,  that means you pick one or the other.

Generally the artifacts we ship are:
*  `eventing-crds.yaml`
* `eventing-core.yaml` 
* `channel-broker.yaml` 
* `mt-channel-broker.yaml`

How about we have the operator to apply the first two yamls, and based on installation choice (or later updated CR), we pick which broker we apply?

The default would be `channel`,

```yaml
apiVersion: operator.knative.dev/v1alpha1
kind: KnativeEventing
metadata:
  name: knative-eventing
  namespace: knative-eventing
spec:
  broker: channel
```

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)**
Which persona is this feature for?

* System integrator

**Exit Criteria**

installation of broker done, based on `broker` field

**Time Estimate (optional):**
How many developer-days do you think this may take to resolve?

**Additional context (optional)**
Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by aliok**
# Issue to be fixed

Fixes #175

## Proposed Changes

* New `DefaultBrokerClass` in KnativeEventing CR
* Make the given broker default broker

Notes:
* Overwrites the `config-br-defaults` configmap based on the value given on CR

TODO:
* Manifest update (waiting until the changes from https://github.com/knative/eventing/pull/2947 updates the file `gs://knative-nightly/eventing/latest/eventing.yaml`)
* Tooling to get new manifest files
* Tests
* Verification instructions

## Release Note

```release-note
```

### CODE REVIEW

File: pkg/reconciler/knativeeventing/controller.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ package knativeeventing
 import (
 	"context"
 	"flag"
+	"k8s.io/client-go/rest"
```
Comment (mattmoor) [inline pkg/reconciler/knativeeventing/controller.go:19]: Format Go code:
```suggestion

```

File: pkg/reconciler/knativeeventing/controller.go [hunk 2]
```diff
@@ -16,6 +16,7 @@ package knativeeventing
...
 	"os"
 	"path/filepath"
```
Comment (mattmoor) [inline pkg/reconciler/knativeeventing/controller.go:21]: Format Go code:
```suggestion
	"path/filepath"

	"k8s.io/client-go/rest"
```

File: pkg/apis/eventing/v1alpha1/knativeeventing_types.go [hunk 3]
```diff
@@ -64,6 +64,11 @@ type KnativeEventingSpec struct {
 	// If no registry is provided, the knative release images will be used.
 	// +optional
 	Registry Registry `json:"registry,omitempty"`
+
+	// The default broker type to use for the brokers Knative creates.
```
Comment (matzew) [inline pkg/apis/eventing/v1alpha1/knativeeventing_types.go:68]: The default broker class ?   I mean, that's what it is :-)

Comment (aliok) [reply pkg/apis/eventing/v1alpha1/knativeeventing_types.go:68]: happy to receive naming suggestions!

File: pkg/reconciler/knativeeventing/controller.go [hunk 4]
```diff
@@ -35,14 +36,17 @@ import (
...
 )
 
 var (
-	recursive  = flag.Bool("recursive", false, "If filename is a directory, process all manifests recursively")
-	MasterURL  = flag.String("master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
-	Kubeconfig = flag.String("kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+	recursive             = flag.Bool("recursive", false, "If filename is a directory, process all manifests recursively")
+	MasterURL             = flag.String("master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+	Kubeconfig            = flag.String("kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+	AvailableBrokerClases = []string{ChannelBasedBrokerClass, MTChannelBasedBrokerClass}
```
Comment (matzew) [inline pkg/reconciler/knativeeventing/controller.go:49]: not sure if available is "right" here.  

There might be others "available" ? I think I still could install my "custom broker", and set its class on the CR, right ? 

Perhaps built-in? :man_shrugging:

File: cmd/manager/kodata/knative-eventing/eventing.yaml [hunk 5]
```diff
@@ -2891,28 +3059,1061 @@ spec:
...
+        - name: CONFIG_OBSERVABILITY_NAME
+          value: config-observability
+        - name: METRICS_DOMAIN
+          value: knative.dev/eventing
+        - # Due to the trivial per-Broker cost, we inject Brokers into every
+          # namespace by default. To change this default simply change this
+          # to "false".  To opt namespaces out of Broker injection, label
+          # them with:
+          #    knative-eventing-injection: disabled
+          name: BROKER_INJECTION_DEFAULT
```
Comment (matzew) [inline cmd/manager/kodata/knative-eventing/eventing.yaml:3699]: why is this here, twice ?

File: cmd/manager/kodata/knative-eventing/eventing.yaml [hunk 6]
```diff
@@ -2891,28 +3059,1061 @@ spec:
...
```
Comment (matzew) [inline cmd/manager/kodata/knative-eventing/eventing.yaml:3985]: and here too

File: pkg/reconciler/knativeeventing/controller.go (hunk 4 above)
Comment (aliok) [reply pkg/reconciler/knativeeventing/controller.go:49]: hmm, never thought of that.
good point. gonna change some things to support it

**Comment by aliok** [comment]
I am rewriting this PR to support customer brokers as well. that will need a different approach
